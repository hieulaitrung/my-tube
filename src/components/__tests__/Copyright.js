import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Copyright from '../Copyright';

describe('<Copyright />', () => {

    test('snapshot renders', () => {
        const component = renderer.create(<Copyright />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders text', () => {
        const { getByText } = render(<Copyright />);
        const textElement = getByText(/Copyright/i);
        expect(textElement).toBeInTheDocument();
    });

    it('should renders date', () => {
        const { getByText } = render(<Copyright />);
        const year = new Date().getFullYear().toString();
        const yearElement = getByText(new RegExp(year, "i"));
        expect(yearElement).toBeInTheDocument();
    });


});