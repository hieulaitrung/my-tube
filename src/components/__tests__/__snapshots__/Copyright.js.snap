// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> snapshot renders 1`] = `
<p
  className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
>
  Copyright Â© 
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
    href="https://material-ui.com/"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    MyTube
  </a>
   
  2020
  .
</p>
`;
